File: src/iedcs-server/security/primes.py
import random, math

class Primes(object):
    global g
    global p
    global secret

    @staticmethod
    def generatePrimes():
        while True:
            p = random.randrange(11, 10000, 2)
            if all(p % n != 0 for n in range(3, int((p ** 0.5) + 1), 2)):
                return p

    @staticmethod
    def generateN(g, p):
        Primes.secret = Primes.generatePrimes()
        tmp = g ** Primes.secret
        return tmp % p

    @staticmethod
    def changePrimes():
        Primes.g = Primes.generatePrimes()
        Primes.p = Primes.generatePrimes()

    @staticmethod
    def generateFinal(p, n):
        tmp =  int(n) ** Primes.secret
        return tmp % p

File: src/iedcs-player/Player/src/player/security/Primes.java
package player.security;

import java.math.BigInteger;
import java.security.SecureRandom;

public class Primes {
    protected static int secret;
    //partially used: http://stackoverflow.com/questions/28846174/generating-two-random-prime-numbers-in-java
    public static int generatePrime(){
        int random;
        SecureRandom r =new SecureRandom();
        
        while (true) {
            random = (int) (r.nextInt() * (127 - 2) + 2);
            if(isPrime(random)){
                break;
            }
        }
        return random;
    }
    
    public static int generateN(int g, int p){
        secret = generatePrime();
        BigInteger tmp;
        tmp = BigInteger.valueOf((long) g);
        long n = tmp.modPow(BigInteger.valueOf((long) secret), BigInteger.valueOf((long) p)).longValue();
        return (int) n;
    }
    
    public static int generateFinal(int p, int n){
        BigInteger tmp;
        tmp = BigInteger.valueOf((long) n);
        long fin = tmp.modPow(BigInteger.valueOf((long) secret), BigInteger.valueOf((long) p)).longValue();
 
        return  (int) fin;
    }
    
    private static boolean isPrime(int n) {
       int i;
       for(i=2;i<=Math.sqrt(n);i++){
           if(n % i == 0){
               return false;
           }
       }
       return true;
    }
}
